### PR TITLE
NumberInputLazy for artifact substat input

### PR DESCRIPTION
**Pull request by chengxi600**
## Describe your changes

Replaced CustomNumberInput with NumberInputLazy

## Issue or discord link

Resolves https://github.com/frzyc/genshin-optimizer/issues/2039
## Testing/validation

https://github.com/frzyc/genshin-optimizer/assets/23622319/f671a1bc-bc72-4450-adfe-5752c66e9acd

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [x] I have made corresponding changes to README or wiki.
- [x] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

File: libs/gi/ui/src/components/artifact/editor/SubstatInput.tsx [hunk 1]
```diff
@@ -129,26 +128,31 @@ export function SubstatInput({
                 </MenuItem>
               ))}
           </DropdownButton>
-          <CustomNumberInputButtonGroupWrapper
-            sx={{ flexBasis: 30, flexGrow: 1 }}
+          <Box
```
Comment (frzyc) [inline libs/gi/ui/src/components/artifact/editor/SubstatInput.tsx:131]: Can you try putting the `NumberInputLazy` here without the `Box` wrapper? You can apply `sx` directly to the `NumberInputLazy`

File: libs/gi/ui/src/components/artifact/editor/SubstatInput.tsx [hunk 2]
```diff
@@ -129,26 +128,31 @@ export function SubstatInput({
...
                 setSubstat(index, { key, value: value ?? 0 })
               }
               disabled={!key}
               error={!!error}
-              sx={{
-                px: 1,
-              }}
               inputProps={{
-                sx: { textAlign: 'right' },
+                sx: { textAlign: 'right', height: '1vh' },
```
Comment (frzyc) [inline libs/gi/ui/src/components/artifact/editor/SubstatInput.tsx:152]: What does the `1vh` here do?

**Review by frzyc — CHANGES_REQUESTED** [review]

File: libs/gi/ui/src/components/artifact/editor/SubstatInput.tsx (hunk 2 above)
Comment (chengxi600) [reply libs/gi/ui/src/components/artifact/editor/SubstatInput.tsx]: was trying to set the size to be what it was before, moved it to the parent Box

**Review by frzyc — CHANGES_REQUESTED** [review]
I assumed this is caused by the `5vh` in height for the parent wrapper
image
I am not sure why you needed to set the height here manually?

**Comment by chengxi600** [comment]
> I assumed this is caused by the `5vh` in height for the parent wrapper image I am not sure why you needed to set the height here manually?

without setting the height it looks like this for me:
Screenshot 2024-05-18 at 12 09 30 PM

with 5vh:
Screenshot 2024-05-18 at 12 10 02 PM

would it be better to just use fixed height or just not set any height?

**Comment by frzyc** [comment]
I think `vh` is fundamentally the wrong unit here, because its a relative unit to the viewheight, which can be different across different screensizes.
The height of the element changes as the view height is changed(by using console)

Screenshot 2024-05-18 212326

image

**Review by frzyc — APPROVED** [review]
LGTM

**Pull request merged**